ext/bson/read.c

```diff
@@ -143,7 +143,7 @@ VALUE rb_bson_byte_buffer_get_bytes(VALUE self, VALUE i)
 }
 
 VALUE pvt_get_boolean(byte_buffer_t *b){
-  VALUE result;
+  VALUE result = 0;
```

Comment:
I don't think you can do this. If you want to silence the compiler warning, you can assign `Qnil` to `result` immediately before the `pvt_raise_decode_error` call in that branch only.

Reply:
And add a comment that the assignment is to pacify the compiler.

Reply:
I just got rid of it.